api/data_explorer/util/elasticsearch_util.py

```diff
@@ -15,59 +18,32 @@
 from flask import current_app
 
 
-def _get_field_min_max_agg(es, field_name):
-    return Search(
-        using=es, index=current_app.config['INDEX_NAME']
-    ).aggs.metric(
-        'max',
-        'max',
-        # Don't execute query; we only care about aggregations. See
-        # https://www.elastic.co/guide/en/elasticsearch/reference/current/returning-only-agg-results.html
-        field=field_name
-    ).params(size=0).aggs.metric(
-        'min',
-        'min',
-        # Don't execute query; we only care about aggregations. See
-        # https://www.elastic.co/guide/en/elasticsearch/reference/current/returning-only-agg-results.html
-        field=field_name).params(size=0).execute()
-
-
-# TODO(bfcrampton): Generalize this for any nested field
-def _get_samples_field_min_max_agg(es, field_name):
-    search = Search(using=es, index=current_app.config['INDEX_NAME'])
-    search.update_from_dict({
-        "aggs": {
-            "parent": {
-                "nested": {
-                    "path": "samples"
-                },
-                "aggs": {
-                    "max": {
-                        "max": {
-                            "field": field_name
-                        }
-                    },
-                    "min": {
-                        "min": {
-                            "field": field_name
-                        }
-                    }
-                }
-            }
-        }
-    })
-    return search.params(size=0).execute()
-
-
 def _get_field_range(es, field_name):
-    if field_name.startswith('samples.'):
-        response = _get_samples_field_min_max_agg(es, field_name)
-        return (response.aggregations.parent['max']['value'] -
-                response.aggregations.parent['min']['value'])
-    else:
-        response = _get_field_min_max_agg(es, field_name)
-        return (response.aggregations['max']['value'] -
-                response.aggregations['min']['value'])
+    search = Search(using=es, index=current_app.config['INDEX_NAME'])
+    # Traverse down the nesting levels from the root field, until we reach the leaf.
+    # Need to traverse until the root, because we have to build the search object
+    # by adding Nested aggregations consecutively. For example, a nested "samples.foo"
+    # field will result in:
+    # Search(...).bucket('samples', Nested(path='samples')).metric(...)
+    parts = field_name.split('.')
+    bucket = search.aggs
+    parent = ''
+    nestings = []
+    for part in parts:
+        parent = '%s.%s' % (parent, part) if parent else part
+        if parent in current_app.config['NESTED_PATHS']:
+            bucket = bucket.bucket(parent, Nested(path=parent))
+            nestings.append(parent)
+
+    bucket.metric('max', Max(field=field_name))
+    bucket.metric('min', Min(field=field_name))
+
+    print 'SEARCH: %s' % search.to_dict()
```

Comment:
Delete?

As an aside, did you see this output? I thought `print`'s don't show up, only `current_app.logger` do.

Reply:
Done.

Hmm I actually did I think, but trying it again it didn't work so maybe not.